src/App/Order/Calculator/PostNl/PostNLReceiptCodeCalculator.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MyParcelNL\Pdk\App\Order\Calculator\PostNl;
+
+use MyParcelNL\Pdk\App\Order\Calculator\AbstractPdkOrderOptionCalculator;
+use MyParcelNL\Pdk\Base\Service\CountryCodes;
+use MyParcelNL\Pdk\Types\Service\TriStateService;
+use MyParcelNL\Pdk\Validation\Validator\CarrierSchema;
+use MyParcelNL\Pdk\Facade\Pdk;
+
+/**
+ * When receipt code is enabled, insurance is required.
+ */
+final class PostNLReceiptCodeCalculator extends AbstractPdkOrderOptionCalculator
+{
+    public function calculate(): void
+    {
+        $shipmentOptions = $this->order->deliveryOptions->shipmentOptions;
+
+        if ($this->order->shippingAddress->cc !== CountryCodes::CC_NL) {
+            $shipmentOptions->receiptCode = TriStateService::DISABLED;
+            return;
+        }
+
+        if (TriStateService::ENABLED === $shipmentOptions->ageCheck) {
+            $shipmentOptions->receiptCode = TriStateService::DISABLED;
+            return;
+        }
+
+        if (TriStateService::ENABLED !== $shipmentOptions->receiptCode) {
+            return;
+        }
+
+        $shipmentOptions->signature     = TriStateService::DISABLED;
+        $shipmentOptions->onlyRecipient = TriStateService::DISABLED;
+        $shipmentOptions->largeFormat   = TriStateService::DISABLED;
+        $shipmentOptions->return        = TriStateService::DISABLED;
+
+        if ($shipmentOptions->insurance <= 1) {
+            /** @var \MyParcelNL\Pdk\Validation\Validator\CarrierSchema $schema */
+            $schema = Pdk::get(CarrierSchema::class);
+            $allowedInsuranceAmounts = $schema
+                ->setCarrier($this->order->deliveryOptions->carrier)
+                ->getAllowedInsuranceAmounts();
+
+            $shipmentOptions->insurance = $this->getMinimumInsuranceAmount($allowedInsuranceAmounts, 10000);
```

Comment:
Ik houd niet zo van onbenoemde "magic numbers" in code. Kun je deze `10000` waarde ook in een constante zetten of op z'n minst voorzien van een comment (waarom dit aantal??)

Reply:
Het liefst dit zo oppakken dat de laagste tier gepakt wordt, ongeacht welk bedrag daar in staat. (En van mij mag het fout gaan als er geen tiers zijn of de laagste tier is 0, want dat is toch een onmogelijke combinatie.)